recipes/agent.rb

```diff
@@ -51,6 +53,13 @@
   chef_environment: node.chef_environment
 }
 
+# set lumberjack key locations and perms
+node.default['lumberjack']['ssl_dir'] = node['logstash']['instance_default']['basedir']
+node.default['lumberjack']['ssl_key_path'] = "#{node['lumberjack']['ssl_dir']}/#{node['lumberjack']['ssl key']}"
+node.default['lumberjack']['ssl_cert_path'] = "#{node['lumberjack']['ssl_dir']}/#{node['lumberjack']['ssl certificate']}"
+node.default['lumberjack']['user'] = node['logstash']['instance_default']['user']
+node.default['lumberjack']['group'] = node['logstash']['instance_default']['group']
+
```

Comment:
I'm always wary of having one default attribute depend on the value of another default attribute; it means when someone overrides one attribute, they have to override everything.

Maybe these could be `node.default_unless['lumberjack']...`?